pkg/pb/metainfo.proto

```diff
@@ -193,3 +213,327 @@ message ProjectInfoRequest {
+    google.protobuf.Timestamp status_at  = 7 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+    google.protobuf.Timestamp expires_at = 8 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+
+    bytes  encrypted_metadata_nonce = 9;
+    bytes  encrypted_metadata       = 10;
+
+    int64                           fixed_segment_size    = 11;
+    pointerdb.RedundancyScheme      redundancy_scheme     = 12;
+    encryption.EncryptionParameters encryption_parameters = 13;
+
+    int64 total_size  = 14;
+    int64 inline_size = 15;
+    int64 remote_size = 16;
+}
+
+message ObjectBeginRequest {
+    bytes  bucket = 1;
+    bytes  encrypted_path = 2;
+    int32  version = 3;
+
+    google.protobuf.Timestamp expires_at = 4 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+
+    bytes  encrypted_metadata_nonce = 5;
+    bytes  encrypted_metadata = 6; // TODO: set maximum size limit
+
+    pointerdb.RedundancyScheme      redundancy_scheme = 7; // can be zero
+    encryption.EncryptionParameters encryption_parameters = 8; // can be zero
+}
+
+message ObjectBeginResponse {
+    bytes  bucket = 1;
+    bytes  encrypted_path = 2;
+    int32  version = 3;
+
+    bytes  stream_id = 4 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+
+    pointerdb.RedundancyScheme      redundancy_scheme = 5;
+    encryption.EncryptionParameters encryption_parameters = 6;
+}
+
+message ObjectCommitRequest {
+    bytes  stream_id = 4 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+}
+
+message ObjectCommitResponse {
+    Object object = 1;
+}
+
+message ObjectListRequest {
+    bytes     bucket = 1;
+    bytes     encrypted_prefix = 2;
+    bytes     encrypted_cursor = 3;
+    int32     limit = 4;
+    bool      recursive = 5;
+
+    ObjectListItemFlags object_flags = 6;
+}
+
+message ObjectListResponse {
+    repeated ObjectListItem items = 1;
+    bool more = 2;
+}
+
+message ObjectListItem {
+    bytes  encrypted_path = 1;
+    int32  version        = 2;
+    Object.Status status  = 3;
+
+    google.protobuf.Timestamp created_at = 4 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+    google.protobuf.Timestamp status_at  = 5 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+    google.protobuf.Timestamp expires_at = 6 [(gogoproto.stdtime) = true, (gogoproto.nullable) = false];
+
+    bytes  encrypted_metadata_nonce = 7;
+    bytes  encrypted_metadata       = 8;
+}
+
+message ObjectListItemFlags {
+    bool metadata = 1;
+}
+
+message ObjectBeginDeleteRequest {
+    bytes  bucket = 1;
+    bytes  encrypted_path = 2;
+    int32  version = 3;
+}
+
+message ObjectBeginDeleteResponse {
+    bytes  stream_id = 1 [(gogoproto.customtype) = "StreamID", (gogoproto.nullable) = false];
+
+    // TODO: should this contain a list of segments needing to be deleted?
+}
+
+message ObjectFinishDeleteRequest {
```

Comment:
Shouldn't we have here only `streamID` similar to `ObjectCommitRequest`? @egonelbre

Reply:
Yeah